### ISSUE TITLE
Features for next major version

### ISSUE THREAD

**Issue by edongashi** [WPF-Forge/Forge.Forms#34]
- Remove dependency on Material Design in XAML and Mahapps.Metro. Create respective packages `Forge.Forms.Material` and `Forge.Forms.Metro`. This means that `Show.Dialog` and `Show.Window` and other utilities won't be part of the core library.
- Target .NET Core 3 WPF - no idea yet how this will work.
- Remove humanizer dependency - handle sentence parser manually.
- Create abstract Form DOM. This dom can be serialized to XML, XAML, Classes, and custom visitors. This dom can be built from XML, Classes, fluent API. Currently `FormElement` is too brittle for that.
- First class XML parser - generalize FormBuilder.
- Make it easier to attach custom controls - placeholders, data templates and data template selectors, return FrameworkElements directly.
- Better support for plugins.
- First class collections support - easy and intuitive API with in-depth customization.

Other suggestions are welcome.

**Comment by redbaty** [comment]
.NET Core 3 port available at https://github.com/WPF-Forge/Forge.Forms/tree/net3

**Comment by edongashi** [comment]
Is it possible to reference .NET framework dependencies (for example NuGet) from a .NET core 3 WPF app?

**Comment by redbaty** [comment]
Yup, it uses wichever version is available for .NET 4.6.1 or beyond, just like .NET Core

**Comment by edongashi** [comment]
That certainly makes migration easier. We will see how it plays out once it's released. By then we will probably get rid of all dependencies from core library anyway.
